pkg/autohealing/cloudprovider/openstack/provider.go

```diff
@@ -253,15 +253,14 @@ func (provider OpenStackCloudProvider) Repair(nodes []healthcheck.NodeInfo) erro
 	if isWorkerNode {
 		for _, n := range nodes {
 			nodesToReplace := sets.NewString()
-			machineID := uuid.Parse(n.KubeNode.Status.NodeInfo.MachineID)
-			if machineID == nil {
+			serverID := strings.TrimPrefix(n.KubeNode.Spec.ProviderID, "openstack:///")
```

Comment:
can we make a util function to get the instance id here? should we consider fall back on `NodeInfo.MachineID` in case `ProviderID` not set?